internal/provider/schema_gitlab_project_milestone.go

```diff
@@ -0,0 +1,121 @@
+package provider
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/xanzy/go-gitlab"
+)
+
+func gitlabProjectMilestoneGetSchema() map[string]*schema.Schema {
+	validMilestoneStates := []string{"active", "closed"}
+
+	return map[string]*schema.Schema{
+		"iid": {
+			Description: "The ID of the milestone only in one project.",
+			Type:        schema.TypeInt,
+			Computed:    true,
+		},
+		"milestone_id": {
+			Description: "The global ID of the project’s milestone.",
+			Type:        schema.TypeInt,
+			Computed:    true,
+		},
+		"project_id": {
```

Comment:
I think that this will eventually lead to problems, because it's an actual field from the GitLab API response which **always** is an integer id of the associated project. Thus, if someone passes in a path with namespace (e.g. in the data source) then there is a config - state conflict. 

I suggest to add an additional `project` argument with the same schema as this one here and use the `project_id` as a `Computed: true` field, only - with the data returned from the GitLab API ... WDYT?

Reply:
@timofurrer 
Mmm, that's a tough question. This will not happen because in Milestone Golang cline the mechanism of work differs from Issues. For example, in `gitlabProjectIssueToStateMap` function we [add project value](https://github.com/gitlabhq/terraform-provider-gitlab/blob/main/internal/provider/schema_gitlab_project_issue.go#L284) and this value [is written in `project`](https://github.com/gitlabhq/terraform-provider-gitlab/blob/main/internal/provider/schema_gitlab_project_issue.go#L286). In my case, this value [is got from response of Golang client](https://github.com/Moon1706/terraform-provider-gitlab/blob/main/internal/provider/schema_gitlab_project_milestone.go#L97) and it's **always** string with project ID even if I contain in options path with namespace.
Probably, we'll have only one problem - person put and expect a path with namespace, but get project ID in state. But I reckon it's okay. If you don't agree, I change it.

Reply:
To demonstrate the problem can you change [this](https://github.com/gitlabhq/terraform-provider-gitlab/pull/1044/files#diff-48bb913f0116bcb2402a3448e78d0167cde6401eb0c523d62b03d82349c4b93fR27) test config to:

```go
               Config: fmt.Sprintf(`
                resource "gitlab_project_milestone" "this" {
                  project_id = "%s"
                  title      = "test-%d"
                }`, project.PathWithNamespace, rInt1),
```

... which is a supported use case. 

When you run this you'll get this error:

```
=== RUN   TestAccGitlabProjectMilestone_basic
    resource_gitlab_project_milestone_test.go:19: Step 1/4 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
        -/+ destroy and then create replacement
        
        Terraform will perform the following actions:
        
          # gitlab_project_milestone.this must be replaced
        -/+ resource "gitlab_project_milestone" "this" {
              ~ created_at   = "2022-05-13T06:53:06Z" -> (known after apply)
              ~ expired      = false -> (known after apply)
              ~ id           = "root/acctest-5013295642796043147:5" -> (known after apply)
              ~ iid          = 1 -> (known after apply)
              ~ milestone_id = 5 -> (known after apply)
              ~ project_id   = "52" -> "root/acctest-5013295642796043147" # forces replacement
              ~ updated_at   = "2022-05-13T06:53:06Z" -> (known after apply)
              ~ web_url      = "http://fd87da6cdd76/root/acctest-5013295642796043147/-/milestones/1" -> (known after apply)
                # (2 unchanged attributes hidden)
            }
        
        Plan: 1 to add, 0 to change, 1 to destroy.
--- FAIL: TestAccGitlabProjectMilestone_basic (3.21s)
```

Note the line `~ project_id   = "52" -> "root/acctest-5013295642796043147" # forces replacement`. 

This is the problem I was referring to. What's going on here is that the API **always** returns an integer `id` no matter if you called the GitLab API using `/projects/52/milestones/1` or `/projects/root%2Facctest-5013295642796043147/milestones/1`. And then always writing this integer id into the `project_id` attribute will result in that state / plan mismatch.

I hope this helped, let me know :)

Reply:
It's strange... Okey, in this case I agree with you and I've added `project` field.